CONTRIBUTING.md

```diff
@@ -1,30 +1,48 @@
-# Contributing
-
-We welcome your contributions. To contribute, please
-- create an [issue](https://github.com/IntelPython/dpctl/issues/new),
-- participate in [discussions](https://github.com/IntelPython/dpctl/discussions), or
-- open a [pull request](https://github.com/IntelPython/dpctl/compare) from changes committed to your own fork
-of `dpctl`.
-
-**N.B.**: Please make sure to tick the checkmark "[x] Allow edits from maintainers" to allow proper functioning
-of automation bots, see "[Working with forks](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)" GitHub document for more details.
-
+# Contributing <!-- omit in toc -->
+
+We welcome your contributions!
+
+To contribute, do one of the following:
+- create an [issue](https://github.com/IntelPython/dpctl/issues/new)
+- participate in [discussions](https://github.com/IntelPython/dpctl/discussions)
+- open a [pull request](https://github.com/IntelPython/dpctl/compare) from changes committed to your fork
+of `dpctl`
+
+> **NOTE:** Make sure to check the box ``"[x] Allow edits from maintainers"`` to allow the proper functioning
+> of automation bots. See [Working with forks](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork) for more details.
+
+# Table of Contents <!-- omit in toc -->
+- [Mechanical Source Issues](#mechanical-source-issues)
+  - [Source Code Formatting](#source-code-formatting)
+    - [C/C++ Code Style](#cc-code-style)
+    - [Python Code Style](#python-code-style)
+    - [Setting Up a Pre-commit](#setting-up-a-pre-commit)
+    - [C/C++ File Headers](#cc-file-headers)
+    - [Python* File Headers](#python-file-headers)
+  - [Security](#security)
+    - [Bandit](#bandit)
+  - [Code Coverage](#code-coverage)
+  - [Error Reporting and Logging](#error-reporting-and-logging)
+    - [Optional use of the Google* logging library (glog*)](#optional-use-of-the-google-logging-library-glog)
 
 
 # Mechanical Source Issues
 
 ## Source Code Formatting
 
-### C/C++ code style
+### C/C++ Code Style
 
-We use [clang-format](https://clang.llvm.org/docs/ClangFormat.html) code formatter.
+We use a [clang-format](https://clang.llvm.org/docs/ClangFormat.html) code formatter.
```

Comment:
```suggestion
We use [clang-format](https://clang.llvm.org/docs/ClangFormat.html) code formatter.
```

"a" does not seem appropriate here (does not pass the replacement with "one" test). We should not use any oarticle to convey a matter-of-fact statement.